proxy/src/error.rs

```diff
@@ -0,0 +1,155 @@
+use juniper::{
+    FieldError, IntoFieldError, Value,
+};
+
+use radicle_registry_client;
+use radicle_registry_client::{Error as RegistryError};
+use radicle_surf::{
+    file_system::{Path},
+};
+
+use crate::source::{Error as SourceError};
+
+#[derive(Debug)]
+/// Enumerable of expected error types.
+pub enum Error {
+    /// File at a given path was irretrievable.
+    FileNotFound(Path),
+    /// Directory at a given path was irretrievable.
+    DirectoryNotFound(Path),
+    /// Project name exceeded 32 characters.
+    BadProjectName(String),
```

Comment:
```suggestion
    ProjectNameValidation(String),
```
This would be my suggestion, disregard if not an improvement.

Reply:
This could hold all possible project name validation errors. Which to start with would be an enum with a single variant.

``` rust
enum ProjectNameValidation {
    TooLong(u32)
}
```

Reply:
This is good. Implemented.